develop/reference/articles/06-themes/02-theme-components-and-workflow.markdown

```diff
@@ -0,0 +1,238 @@
+  for the navigation menu. Note that `navigation.ftl` also defines the hamburger 
+  icon and `navbar-collapse` class that provides the simplified navigation 
+  toggle for mobile viewports, as shown in the snippet below for the Classic 
+  theme:
+
+      <#if has_navigation && is_setup_complete>
+        <button aria-controls="navigationCollapse" aria-expanded="false" 
+        aria-label="Toggle navigation" class="navbar-toggler navbar-toggler-right" 
+        data-target="#navigationCollapse" data-toggle="collapse" type="button">
+          <span class="navbar-toggler-icon"></span>
+        </button>
+
+        <div aria-expanded="false" class="collapse navbar-collapse" id="navigationCollapse">
+          <@liferay.navigation_menu default_preferences="${preferences}" />
+        </div>
+      </#if>
+
+![Figure 1: The collapsed navbar provides simplified user-friendly navigation for mobile devices.](../../images/portal-layout-mobile-nav.png)
+
+- `portal_pop_up.ftl`: The theme template for pop up dialogs for the theme's 
+  portlets. Similar to `portal_normal.ftl`, `portal_pop_up.ftl` provides the 
+  markup template for all pop-up dialogs, such as a portlet's Configuration menu 
+  (includes `init.ftl`, which includes `init_custom.ftl`).
+
+![Figure 2: Each theme template provides a portion of the page's markup and functionality.](../../images/portal-layout-theme-templates.png)
+
+### Theme Template Utilities [](id=theme-template-utilities)
+
+Liferay provides several FreeMarker variables and macros that you can use in 
+your theme templates to include portlets, use taglibs, access theme objects, and 
+more. You can see examples of these used in `portal_normal`. These utilities are 
+included in the files listed below:
+
+- [`Init.ftl`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/frontend-theme/frontend-theme-unstyled/src/main/resources/META-INF/resources/_unstyled/templates/init.ftl): 
+  Provides access to common theme variables
+- [`FTL_Liferay.ftl`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl): 
+  Provides macros for commonly used portlets and theme resources. See the 
+  [Macros tutorial](/develop/tutorials/-/knowledge_base/7-1/using-liferays-macros-in-your-theme) 
+  for more information.
+- [`taglib-mappings.properties`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/frontend-taglib/frontend-taglib-util-freemarker-contributor/src/main/resources/META-INF/taglib-mappings.properties): 
+  Maps the portal taglibs to names for FreeMarker macros. Taglibs let you 
+  quickly create common UI components. This properties file is also provided 
+  separately for each app taglib. See the 
+  [Taglib tutorials](/develop/tutorials/-/knowledge_base/7-1/front-end-taglibs) 
+  for more information on using each taglib in your theme templates.
+
```

Comment:
I don't think the links to the source code are so much useful here. It would be good if we could actually collect what is available in either a reference related document or inline here so developers don't need to dig through the source code.

Reply:
Good point. I can definitely add it here. Thanks!

Reply:
I opted to add theme a separate reference guide since there were so many. See the latest commit for the updates. Thanks!